packages/engine/Specs/Scene/Model/VerticalExaggerationPipelineStageSpec.js

```diff
@@ -0,0 +1,53 @@
+import {
+  Cartesian2,
+  RenderState,
+  ShaderBuilder,
+  VerticalExaggerationPipelineStage,
+} from "../../../index.js";
+import ShaderBuilderTester from "../../../../../Specs/ShaderBuilderTester.js";
+
+describe(
+  "Scene/Model/VerticalExaggerationPipelineStage",
+  function () {
+    const mockModel = {};
+    const mockPrimitive = {};
+    const mockFrameState = {
+      verticalExaggeration: 2.0,
+      verticalExaggerationRelativeHeight: 100.0,
+    };
+
+    function mockRenderResources() {
+      return {
+        model: mockModel,
+        shaderBuilder: new ShaderBuilder(),
+        uniformMap: {},
+        renderStateOptions: RenderState.getState(RenderState.fromCache()),
+      };
+    }
+
+    it("adds defines and uniforms", function () {
```

Comment:
I would also add a check for `ShaderBuilderTester.expectVertexLinesEqual(shaderBuilder, [_shadersVerticalExagerationStageVS]);`

:thinking: I also notice that not every pipeline stage spec checks if the shader code was added. Was that an omission in the past, or has anything changed in the past year?

Reply:
I added the check for the relevant shader code. I'm not sure why some pipeline stage specs don't check this--we haven't consciously decided not to.